apps/extension/src/background/keyring/keyring.ts

```diff
@@ -630,16 +599,59 @@ export class KeyRing {
     return accounts.map((account) => account.public as AccountStore);
   }
 
+  /**
+   * For provided address, return associated private key
+   */
+  private async getSigningKey(address: string): Promise<string> {
+    const account = await this.vaultService.findOne<AccountStore>(
+      KEYSTORE_KEY,
+      "address",
+      address
+    );
+    if (!account) {
+      throw new Error(`Account for ${address} not found!`);
+    }
+    const accountStore = (await this.queryAllAccounts()).find(
+      (account) => account.address === address
+    );
+
+    if (!accountStore) {
+      throw new Error(`Account for ${address} not found!`);
+    }
+    const { path } = accountStore;
+    const { coinType } = chains[defaultChainId].bip44;
+    const bip44Path = makeBip44PathArray(coinType, path);
+
+    const sensitiveProps =
+      await this.vaultService.reveal<SensitiveAccountStoreData>(account);
+    if (!sensitiveProps) {
+      throw new Error(`Signing key for ${address} not found!`);
+    }
+    const { text: phrase } = sensitiveProps;
+    const mnemonic = Mnemonic.from_phrase(phrase);
+    const seed = mnemonic.to_seed();
+    const hdWallet = new HDWallet(seed);
+    const key = hdWallet.derive(new Uint32Array(bip44Path));
+    const privateKeyStringPtr = key.to_hex();
```

Comment:
I think you should free the pointer at some point right? So private key zeroizes.

Reply:
Done!